Sources/WeatherRecorder/WeatherRecorder.swift

```diff
@@ -70,34 +70,9 @@ open class WeatherRecorder : NSObject, AsyncActionController, CLLocationManagerD
 
     /// Override to request GPS and motion permissions.
     public func requestPermissions(on viewController: Any, _ completion: @escaping AsyncActionCompletionHandler) {
-        
-        // Get the current status and exit early if the status is restricted or denied.
-        let status = LocationAuthorization.shared.authorizationStatus(for: StandardPermissionType.locationWhenInUse.identifier)
-        if status == .denied || status == .restricted {
-            _serviceFailed(status: status)
-            completion(self, nil, self.error)
-            return
-        }
-        
-        // If the status is not determined, then need to request it.
-        self._setupLocationManager()
-        if status == .notDetermined {
-            self.status = .requestingPermission
-            self._permissionCompletion = completion
-            #if os(macOS)
-            self.locationManager!.requestAlwaysAuthorization()
-            #else
-            self.locationManager!.requestWhenInUseAuthorization()
-            #endif
-        }
-        else {
-            self.status = .permissionGranted
-            completion(self, nil, nil)
-        }
+        self.status = .permissionGranted
```

Comment:
?? Is this no longer where the app requests these permissions from iOS?

Reply:
So... yeah. The recorders aren't meant to run headless so they are expected to ask for permissions *after* the task starts and shows at least the first screen but *before* they are used, and they are intended to be integrated with a task that knows they are there and when is a "good time" to ask for authorization. (Like, after explaining to the user why you want it.) 

Because the task that runs the recorders is just a wrapper for the MTB task, there is nothing to show to the user while the MTB assessment is loading its assets. With some of their tasks, this takes a while b/c they are getting large images and/or audio on the main thread. The main thread dispatches a request for permissions on the main thread next loop, but their views sometimes take a *very* long time to load. The main thread will hang and then the location manager (which also runs on the main thread) doesn't finish asking for permission until *after* they've finished loading their views. Getting all the permissions (if asking for more than one at the start of the task) uses a dispatch group on a background thread to allow all the permissions to be asked for without blocking UI and then there is a timer that waits until the dispatch group is done or a timeout is reached. This was crashing the app.

My work-around is to just let this thing run in the background and do it's thing. Normally, this would result in the permissions being swallowed by screen transitions but b/c we ask for permissions during onboarding this won't happen.

In short, no, the recorder assumes that you have previously asked for permissions or that it's "ok" to ask for them after transitioning to the "next" screen when the recorder is started (or in this case, the *first* screen because our task is running headless). This is a problem for tasks where we are asking the participant to put the device in their pocket, but for this, it's fine.